src/utils/graceful_shutdown.rs

```diff
@@ -0,0 +1,208 @@
+//! A class to handle shutdown signals and to shut down system
+//! Actix system has to be running for it to work.
+
+use std::{
+    collections::{BTreeMap, HashSet},
+    mem,
+    time::Duration,
+};
+
+use actix::{
+    self,
+    prelude::{Actor, Context},
+    Addr, AsyncContext, Handler, Message, Recipient, ResponseActFuture, System,
+    WrapFuture,
+};
+
+use tokio::prelude::{
+    future::{self, join_all, Future},
+    stream::*,
+    FutureExt,
+};
+
+use crate::log::prelude::*;
+
+pub type ShutdownMessageResult =
+    Result<Box<(dyn Future<Item = (), Error = ()> + std::marker::Send)>, ()>;
```

Comment:
@tyranron,

I cannot find a way to send `Box<dyn Future<Item = (), Error = ()>>` as actor's response to `ShutdownMessage` because trait [MessageResponse](https://docs.rs/actix/0.8.3/actix/dev/trait.MessageResponse.html) is not implemented for `Box<...>`

I can try to implement it myself in shutdown.rs

Reply:
@nWacky see `signalling::room` implementation. It doesn't use `Result<...>` wrapping for future. So should you be able to.